src/contact-author.ts

```diff
@@ -0,0 +1,75 @@
+import * as sourcegraph from 'sourcegraph'
+import { EMPTY, from, Observable } from 'rxjs'
+import { filter, map, switchMap } from 'rxjs/operators'
+import { Hunk, queryBlameHunks, resolveURI } from './blame'
+
+const decorationType = sourcegraph.app.createDecorationType()
+
+function observeActiveCodeEditorChanges(): Observable<sourcegraph.CodeEditor> {
+    return from(sourcegraph.app.activeWindowChanges).pipe(
+        switchMap(activeWindow => activeWindow?.activeViewComponentChanges || EMPTY),
+        filter((viewer): viewer is sourcegraph.CodeEditor => !!viewer && viewer.type === 'CodeEditor')
+    )
+}
+
+function observeCodeEditorSelectionChanges(): Observable<{
+    editor: sourcegraph.CodeEditor
+    selections: sourcegraph.Selection[]
+}> {
+    return observeActiveCodeEditorChanges().pipe(
+        switchMap(editor => from(editor.selectionsChanges).pipe(map(selections => ({ editor, selections }))))
+    )
+}
+
+function renderDecorationContent(authorPerson: Hunk['author']['person']): string {
+    return `📣 Contact author: ${authorPerson.displayName}`
+}
+
+function renderMailtoLink(email: string, body: string, subject: string) {
+    return `mailto:${email}?subject=${encodeURIComponent(subject)}&body=${encodeURIComponent(body)}`
+}
+
+function getLineFromText(text: sourcegraph.TextDocument['text'] = '', lineNumber: number): string {
```

Comment:
**question for @felixfbecker**: Is there a reason we don't expose `_lines` on `ExtensionDocument`?

Reply:
Internally, we account for carriage returns as well (for Windows), so it's unfortunate that authors have to re-implement this, as it's easy to forget